clients/da-vinci-client/src/main/java/com/linkedin/davinci/blob/ServerBlobFinder.java

```diff
@@ -0,0 +1,54 @@
+package com.linkedin.davinci.blob;
+
+import com.linkedin.venice.exceptions.VeniceException;
+import com.linkedin.venice.helix.HelixCustomizedViewOfflinePushRepository;
+import com.linkedin.venice.meta.Instance;
+import com.linkedin.venice.meta.Partition;
+import com.linkedin.venice.meta.Version;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+
+public class ServerBlobFinder implements BlobFinder {
+  private final HelixCustomizedViewOfflinePushRepository customizedViewRepository;
+
+  private static final Logger LOGGER = LogManager.getLogger(ServerBlobFinder.class);
+
+  public ServerBlobFinder(CompletableFuture<HelixCustomizedViewOfflinePushRepository> customizedViewRepository) {
+    this.customizedViewRepository = customizedViewRepository.join();
+  }
+
+  @Override
+  public BlobResponse findBlob(String storeName, int version, int partitionId) {
+    BlobResponse response = new BlobResponse();
+    try {
+      String currentVersionResource = Version.composeKafkaTopic(storeName, version);
+      // Get the partition assignments for the current version
+      List<String> partitionValues = new ArrayList<>();
+      for (Partition partition: customizedViewRepository.getPartitionAssignments(currentVersionResource)
+          .getAllPartitions()) {
+        if (partition.getId() == partitionId) {
+          for (Instance instance: partition.getReadyToServeInstances()) {
+            String host = instance.getHost();
+            partitionValues.add(host);
+          }
+          break;
+        }
+      }
+      response.setPartitionValues(partitionValues);
```

Comment:
```suggestion
      String topic = Version.composeKafkaTopic(storeName, version);
      Partition partition = customizedViewRepository.getPartition(topic);
      if (partition != null) {
        response.addPartition(partition);
      }
```
Does this work? (`addPartition()` would need to be implemented)

Reply:
Managed to implement the abstraction